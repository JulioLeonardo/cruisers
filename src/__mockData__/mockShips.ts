import { Ship } from "../types";

const MOCK_SHIPS: Ship[] = [
  {
    __typename: "Ship",
    id: "GOMSTREE",
    active: true,
    home_port: "Port Canaveral",
    name: "GO Ms Tree",
    type: "High Speed Craft",
    image: "https://i.imgur.com/MtEgYbY.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:3439091/vessel:MR%20STEVEN",
    year_built: 2015,
  },
  {
    __typename: "Ship",
    id: "GOPURSUIT",
    active: false,
    home_port: "Port Canaveral",
    name: "GO Pursuit",
    type: "Cargo",
    image: "https://i.imgur.com/5w1ZWre.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:439594/vessel:GO%20PURSUIT",
    year_built: 2007,
  },
  {
    __typename: "Ship",
    id: "AMERICANCHAMPION",
    active: false,
    home_port: "Port of Los Angeles",
    name: "American Champion",
    type: "Tug",
    image: "https://i.imgur.com/woCxpkj.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:434663/vessel:AMERICAN%20CHAMPION",
    year_built: 1976,
  },
  {
    __typename: "Ship",
    id: "GOQUEST",
    active: true,
    home_port: "Port Canaveral",
    name: "GO Quest",
    type: "Cargo",
    image: "https://i.imgur.com/ABXtHKa.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:450521/vessel:GO%20QUEST",
    year_built: 2014,
  },
  {
    __typename: "Ship",
    id: "GONAVIGATOR",
    active: true,
    home_port: "Port Canaveral",
    name: "GO Navigator",
    type: "Cargo",
    image: "https://i.imgur.com/MjNWzhO.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:452540/vessel:GO%20NAVIGATOR",
    year_built: 2009,
  },
  {
    __typename: "Ship",
    id: "JRTI-2",
    active: true,
    home_port: "Port of Los Angeles",
    name: "Just Read The Instructions 2",
    type: "Barge",
    image: "https://i.imgur.com/7VMC0Gn.jpg",
    url: "",
    year_built: 2015,
  },
  {
    __typename: "Ship",
    id: "NRCQUEST",
    active: true,
    home_port: "Port of Los Angeles",
    name: "NRC Quest",
    type: "Cargo",
    image: "https://i.imgur.com/GvtuYUN.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:447294/vessel:NRC%20QUEST",
    year_built: 2002,
  },
  {
    __typename: "Ship",
    id: "OCISLY",
    active: true,
    home_port: "Port Canaveral",
    name: "Of Course I Still Love You",
    type: "Barge",
    image: "https://i.imgur.com/28dCx6G.jpg",
    url: "",
    year_built: 2015,
  },
  {
    __typename: "Ship",
    id: "PACIFICWARRIOR",
    active: false,
    home_port: "Port of Los Angeles",
    name: "Pacific Warrior",
    type: "Tug",
    image: "https://i.imgur.com/qdNDmYs.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:434121/vessel:PACIFIC%20WARRIOR",
    year_built: 1974,
  },
  {
    __typename: "Ship",
    id: "PACIFICFREEDOM",
    active: true,
    home_port: "Port of Los Angeles",
    name: "Pacific Freeedom",
    type: "Tug",
    image: "https://i.imgur.com/eJWx70Z.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:4249791/vessel:PACIFIC%20FREEDOM",
    year_built: 1976,
  },
  {
    __typename: "Ship",
    id: "BETTYRGAMBARELLA",
    active: false,
    home_port: "Port of Los Angeles",
    name: "Betty R Gambarella",
    type: "Tug",
    image: "https://i.imgur.com/ngYgFnn.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:428415/vessel:BETTY%20R%20GAMBARELLA",
    year_built: 1974,
  },
  {
    __typename: "Ship",
    id: "AMERICANISLANDER",
    active: false,
    home_port: "Port of Los Angeles",
    name: "American Islander",
    type: "Cargo",
    image: "https://i.imgur.com/jmj8Sh2.jpg",
    url: "https://www.marinetraffic.com/en/ais/details/ships/shipid:435112/vessel:AMERICAN%20ISLANDER",
    year_built: 1900,
  },
];

export { MOCK_SHIPS };
