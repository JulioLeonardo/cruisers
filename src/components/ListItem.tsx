import React from "react";
import { Ship } from "../types";
import { Box, Card, Link, Typography } from "@mui/material";
import { InfoChip } from "./";
import {
  Circle,
  Flag,
  Foundation,
  Image,
  Mouse,
  TitleOutlined,
} from "@mui/icons-material";
import { useShipsStore } from "../store";

const ListItem = ({ ship }: { ship: Ship }): JSX.Element => {
  const { toggleImageModal } = useShipsStore();

  return (
    <Card sx={{ padding: 3, marginBottom: 1 }}>
      <Typography variant="h5" sx={{ paddingBottom: 1 }}>
        {ship.name}
      </Typography>
      <Box display="flex" gap={2}>
        <InfoChip
          title="Status"
          icon={
            <Circle
              color={ship.active ? "success" : "error"}
              sx={{ width: (theme) => theme.spacing(1.8) }}
            />
          }
          label={ship.active ? "Active" : "Inactive"}
        />
        <InfoChip
          title="Type"
          icon={
            <TitleOutlined
              color="secondary"
              sx={{ width: (theme) => theme.spacing(2.5) }}
            />
          }
          label={ship.type}
        />
        <InfoChip
          title="Home port"
          icon={
            <Flag color="info" sx={{ width: (theme) => theme.spacing(2.5) }} />
          }
          label={ship.home_port}
        />
        <InfoChip
          title="Year built"
          icon={
            <Foundation
              color="action"
              sx={{ width: (theme) => theme.spacing(2.5) }}
            />
          }
          label={String(ship.year_built)}
        />
        <Box onClick={() => toggleImageModal(String(ship.id))}>
          <InfoChip
            title="Image"
            icon={
              <Image
                color="primary"
                sx={{ width: (theme) => theme.spacing(2.5) }}
              />
            }
            label="Photo"
          />
        </Box>
        <Link target="_blank" href={ship.url}>
          <InfoChip
            title="Link - Marine Traffic"
            icon={
              <Mouse
                color="action"
                sx={{ width: (theme) => theme.spacing(2.5) }}
              />
            }
            label="Marine Traffic - Ship page"
          />
        </Link>
      </Box>
    </Card>
  );
};

export { ListItem };
