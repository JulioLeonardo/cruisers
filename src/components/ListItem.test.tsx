import React from "react";
import TestRenderer from "react-test-renderer";
import { ListItem } from "./ListItem";
import { MOCK_SHIPS } from "../__mockData__/mockShips";

it("renders correctly", () => {
  const tree = TestRenderer.create(<ListItem ship={MOCK_SHIPS[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
