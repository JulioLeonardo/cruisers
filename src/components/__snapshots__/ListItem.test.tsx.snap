// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-qxzq5-MuiPaper-root-MuiCard-root"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-144bb4j-MuiTypography-root"
  >
    GO Ms Tree
  </h5>
  <div
    className="MuiBox-root css-1dqpc7h"
  >
    <div
      aria-label="Status"
      aria-labelledby={null}
      className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1ytpwe4-MuiChip-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorSuccess css-1hsi6dz-MuiSvgIcon-root"
        data-testid="CircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
        />
      </svg>
      <span
        className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Active
      </span>
    </div>
    <div
      aria-label="Type"
      aria-labelledby={null}
      className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1ytpwe4-MuiChip-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorSecondary css-ql42lq-MuiSvgIcon-root"
        data-testid="TitleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 4v3h5.5v12h3V7H19V4H5z"
        />
      </svg>
      <span
        className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        High Speed Craft
      </span>
    </div>
    <div
      aria-label="Home port"
      aria-labelledby={null}
      className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1ytpwe4-MuiChip-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorInfo css-hjc7um-MuiSvgIcon-root"
        data-testid="FlagIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.4 6 14 4H5v17h2v-7h5.6l.4 2h7V6z"
        />
      </svg>
      <span
        className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        Port Canaveral
      </span>
    </div>
    <div
      aria-label="Year built"
      aria-labelledby={null}
      className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1ytpwe4-MuiChip-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorAction css-1950t6l-MuiSvgIcon-root"
        data-testid="FoundationIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 12h3L12 3 2 12h3v3H3v2h2v3h2v-3h4v3h2v-3h4v3h2v-3h2v-2h-2v-3zM7 15v-4.81l4-3.6V15H7zm6 0V6.59l4 3.6V15h-4z"
        />
      </svg>
      <span
        className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
      >
        2015
      </span>
    </div>
    <div
      className="MuiBox-root css-13o7eu2"
      onClick={[Function]}
    >
      <div
        aria-label="Image"
        aria-labelledby={null}
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1ytpwe4-MuiChip-root"
        data-mui-internal-clone-element={true}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary css-1bs1du6-MuiSvgIcon-root"
          data-testid="ImageIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
          />
        </svg>
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          Photo
        </span>
      </div>
    </div>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://www.marinetraffic.com/en/ais/details/ships/shipid:3439091/vessel:MR%20STEVEN"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      <div
        aria-label="Link - Marine Traffic"
        aria-labelledby={null}
        className="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1ytpwe4-MuiChip-root"
        data-mui-internal-clone-element={true}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorAction css-1950t6l-MuiSvgIcon-root"
          data-testid="MouseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 1.07V9h7c0-4.08-3.05-7.44-7-7.93zM4 15c0 4.42 3.58 8 8 8s8-3.58 8-8v-4H4v4zm7-13.93C7.05 1.56 4 4.92 4 9h7V1.07z"
          />
        </svg>
        <span
          className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
        >
          Marine Traffic - Ship page
        </span>
      </div>
    </a>
  </div>
</div>
`;
